sympy/sets/sets.py

```diff
@@ -1202,28 +1202,32 @@ def reduce(args):
 
         # ===== Pair-wise Rules =====
         # Here we depend on rules built into the constituent sets
-        args = set(args)
+        args = FiniteSet(*args)
```

Comment:
Thanks @hargup . I did this changes because of [problem I mentioned](https://github.com/sympy/sympy/pull/11188#issuecomment-229264778) After this changes (using finiteset, instead of list to always get elements in one order) test cases passes(but still I felt there is something that causes the problem). I will try some alternate way.